lib/src/main/java/com/auth0/android/lock/LockActivity.java

```diff
@@ -552,11 +584,16 @@ public void run() {
                     deliverSignUpResult(user);
                 }
             });
+            lastDatabaseSignUp = null;
         }
 
         @Override
         public void onFailure(final AuthenticationException error) {
             Log.e(TAG, "Failed to create the user: " + error.getMessage(), error);
+            if (error.isVerificationRequired()) {
+                completeDatabaseAuthenticationOnBrowser();
```

Comment:
Why on this error callback the `completeDatabaseAuthenticationOnBrowser()` method is called before (outside) the runnable, and in the other error handler (in `authCallback`) it's called inside the runnable?

Reply:
It's actually better to move it outside of the runnable. 👍 

The reason that runnable exists is that the UI is touched on the callback, and it needs to be touched in the main/UI thread. The UI was being updated to hide the progress bar, but it actually makes sense to keep showing it until it comes back from the browser (either with a good callback or because the user canceled the auth). I'll make this change.

Reply:
Moved out of the runnable